lib/services/pool/pool-weighted.service.ts

```diff
@@ -98,7 +98,7 @@ export class PoolWeightedService implements PoolService {
         const userData = this.encodeJoinPool(data);
 
         if (data.zapIntoMasterchefFarm && this.pool.staking?.type === 'MASTER_CHEF' && this.pool.staking.farm) {
```

Comment:
this check is no longer adequate then, should be: `(farm && zapIntoFarm) || (gauge && zapIntoGauge)`

Reply:
Changed it!